db/src/types.rs

```diff
@@ -97,4 +103,7 @@ pub struct TxReport {
     /// existing entid, or is allocated a new entid.  (It is possible for multiple distinct string
     /// literal tempids to all unify to a single freshly allocated entid.)
     pub tempids: BTreeMap<String, Entid>,
+
+    // A set of entids for attributes that were affected inside this transaction
+    pub changeset: AttributeSet,
```

Comment:
I confess that I still don't like this. It's over-specific (why don't we track the types? or the entities? or the values?), 99% of the time it won't be used, and we can do less work and not allocate just by moving checks around.

Did you disagree with my comment the other day, or do you simply want to get this landed and iterate?

Reply:
No, I just forgot to mention that I am planning on tackling this under this issue:
https://github.com/mozilla-prototypes/sync-storage-prototype/issues/55